lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1567,11 +1567,11 @@ protected function _validateAndCompleteOneToOneMapping(array $mapping)
             foreach ($mapping['joinColumns'] as &$joinColumn) {
                 if ($mapping['type'] === self::ONE_TO_ONE && ! $this->isInheritanceTypeSingleTable()) {
                     if (count($mapping['joinColumns']) == 1) {
-                        if ( ! isset($mapping['id']) || ! $mapping['id']) {
-                            $joinColumn['unique'] = true;
+                        if ( (! isset($mapping['id']) || ! $mapping['id'])) {
```

Comment:
That must have been because I had made some changes to that line that I later  removed, but the extra braces must have remained and I didn't notice.